### PR TITLE
Adding CSV string import function

### PR DESCRIPTION
**Pull request by IshaGupta18**
@jywarren @namangupta01, in response to the comment https://github.com/publiclab/simple-data-grapher/issues/26#issuecomment-493991729, here's importing a CSV as a string, please give your reviews. Mentors, please give your inputs too. Thank you!

stringCSV

### CODE REVIEW

File: src/parsing.js [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+	console.log("at image");
+	var x=new Date();
+	var timestamp=x.getTime();
+	$("#save_as_image").click(function() {
+	  $("#canvas").get(0).toBlob(function(blob) {
+	    saveAs(blob, "chart"+timestamp);
+	  });
+	});
+}
+function colorBackground(ctx,canv){
```
Comment (jywarren) [inline src/parsing.js]: Hi! This is starting to mix in view code with parsing code, and it seems like maybe we need to put all views code in its own file, to follow an MVC model. What do you think?

**Comment by jywarren** [comment]
This is looking really good. However, I want to encourage you to take a breather now both @namangupta01 and @IshaGupta18 and consider doing some code reorganizing to put views code in a different place from data processing code and that in a different place from data import code. Ideally, if you standardize the data format, you should be able to pass a standard data from any import source, to the processor, which does stuff, then passes a standard data format to the renderer (the view) for display. Does this make sense? It's easier said than done! But at 400+ lines, your code is now complex enough that it could use this kind of organization, and you'll benefit later in the project! 

Thanks!!!

**Comment by IshaGupta18** [comment]
Yes, I absolutely agree with you. This code does need some refactoring and
reorganizing. So basically, to do this, I should separate thr script for
data parsing and views right? Basically, pass the final data produced after
the parsing functions to the view. Please correct me if I am wrong and
also, what should be a good practice in this case. Thanks a lot!

**Comment by namangupta01** [comment]
As we also have to support mulitple graphs from multiple csvs. I was
thinking of using classes. And divide the work between various instance
methods. What do you think?

**Comment by jywarren** [comment]
Can we try to consult with @sagarpreet-chadha and @rexagod a bit on this? I
think it's a great moment to connect with others looking to organize their
code!

**Comment by sagarpreet-chadha** [comment]
MVC model is undoubtedly easy to grasp and understand . I think the distribution of files in all our projects should follow MVC guidelines .

Now if you will see [PublicLab.editor ](https://github.com/publiclab/PublicLab.Editor) project , the classes (and class inheritance) are beautifully implemented in each module of the editor . I think use of classes there made the code very organized , but if you would see carefully the UI code in mixed with data-processing code in each module , right ? However as we have made various independent modules following same rules/functions to display UI , the code is easy to understand . 

Other approach is implemented in [LBLD](https://github.com/publiclab/leaflet-blurred-location-display) , as @jywarren mentioned earlier we have distributed different UI code in a different folder and 
the data-processing code in the main folder . I think if a project is in its initial stage --- we should follow this approach . Later when we have the basic functionalities implemented , then it will be quite easy to move code around to achieve better architecture . 
UI files : https://github.com/publiclab/leaflet-blurred-location-display/tree/main/src/ui
data-processing file : https://github.com/publiclab/leaflet-blurred-location-display/blob/main/src/blurredLocationDisplay.js

Thank You 😄 !

**Comment by IshaGupta18** [comment]
That's some pretty great advice! I will look into the projects deeper and
see how it can be applied here. Thanks a lot!

**Comment by IshaGupta18** [comment]
Okay so I am reading up about MVC structure in js, it is a little tough to understand, but I will try and figure it out. I was also looking at the LBLD structure, which is very similar to this code, and only the UI based code needs to be separated out. What do you think should be done for now? MVC would definitely be a better choice, but it might take a little longer to implement. @jywarren @namangupta01

**Comment by IshaGupta18** [comment]
Okay so after a lot of debugging and trial and errors, I have fixed the hovering error of the add graph feature (#24) and this is the latest PR having the latest UI design and all the features till now. I suggest merging this PR so that now, the MVC work can be done better. It is really important to merge this PR because all the latest code is contained in it. @jywarren @namangupta01 @gauravano kindly respond soon. Thanks a lot!

**Comment by jywarren** [comment]
OK, I'm good with this if you all are -- thanks a lot! Maybe with a review from @namangupta01 we can then go ahead and merge this?

**Comment by namangupta01** [comment]
This seems good to me.

**Pull request merged**